Cargo.toml

```diff
@@ -4,30 +4,83 @@ version = "0.1.0"
 authors = ["Ogier Bouvier <ogier@bouvier.family>", "Valérian Rousset <tharvik@users.noreply.github.com>"]
 edition = "2018"
 license = "AGPL-3"
+resolver = "2"
 
 [dependencies]
-drop = { git = "https://github.com/Distributed-EPFL/drop.git" }
-murmur = { git = "https://github.com/Distributed-EPFL/murmur.git" }
-sieve = { git = "https://github.com/Distributed-EPFL/sieve.git" }
-contagion = { git = "https://github.com/Distributed-EPFL/contagion.git" }
+drop = { git = "https://github.com/Distributed-EPFL/drop" }
+sieve = { git = "https://github.com/Distributed-EPFL/sieve" }
 bincode = "1.3.3"
-futures = "0.3"
-num_cpus = "1.13"
-prost = "0.7"
+http = "0.2"
+prost = { version = "0.8", default-features = false }
 serde = { version = "1", features = ["derive"] }
-serde_str = "0.1"
 snafu = "0.6"
-structopt = "0.3"
-tokio = { version = "1", features = ["macros", "net", "rt-multi-thread", "sync", "time"] }
-toml = "0.5"
-tonic = "0.4"
-tracing = "0.1"
-tracing-fmt = "0.1"
-url = "2.2"
+tonic = { version = "0.5", default-features = false, features = ["codegen", "prost"] }
+tokio = { version = "1", features = ["macros", "sync", "time"] }
+
+# cli
+hex = { version = "0.4", features = ["serde"], optional = true }
+structopt = { version = "0.3", optional = true }
+toml = { version = "0.5", optional = true }
+
+# client
+serde_str = { version = "0.1", optional = true }
+url = { version = "2.2", optional = true }
+
+# server
+contagion = { git = "https://github.com/Distributed-EPFL/contagion", optional = true }
+futures = { version = "0.3", optional = true }
+murmur = { git = "https://github.com/Distributed-EPFL/murmur", optional = true }
+num_cpus = { version = "1.13", optional = true }
+tonic-web = { version = "0.1", optional = true }
+tracing = { version = "0.1", optional = true }
+tracing-fmt = { version = "0.1", optional = true }
+
+[target.'cfg(not(target_family = "wasm"))'.dependencies]
+tonic = { version = "0.5", default-features = false, features = ["transport"] }
+
+[target.'cfg(target_family = "wasm")'.dependencies]
+grpc-web-client = { git = "https://github.com/titanous/grpc-web-client", branch = "main" }
 
 [dev-dependencies]
 duct = "0.13"
-nix = "0.20"
+nix = "0.23"
 
 [build-dependencies]
-tonic-build = "0.4"
+tonic-build = { version = "0.5", default-features = false, features = ["prost"] }
+
+[features]
+default = []
+_cli = [
+	"drop/system",
+	"hex",
+	"sieve/system",
+	"structopt",
+	"tokio/net",
+	"tokio/rt-multi-thread",
+	"toml",
+]
+client = [
+	"_cli",
```

Comment:
fomyruk: why is there an underscore for the `_cli`?

Reply:
I'm trying to have a "private" feature, to avoid duplicating too much lines. Using an underscore looked like a good way to do so.